apps/stripe/src/app/api/saleor/transaction-initialize-session/use-case.test.ts

```diff
@@ -0,0 +1,101 @@
+import { err, ok } from "neverthrow";
+import { describe, expect, it, vi } from "vitest";
+
+import { mockedStripeConfig } from "@/__tests__/mocks/stripe-config";
+import { MockedStripePaymentIntentsApi } from "@/__tests__/mocks/stripe-payment-intents-api";
+import { mockedTransactionInitializeSessionEvent } from "@/__tests__/mocks/transaction-initalize-session-event";
+import { InitializeStripeTransactionUseCase } from "@/app/api/saleor/transaction-initialize-session/use-case";
+import { BaseError } from "@/lib/errors";
+import { AppConfigPersistor } from "@/modules/app-config/app-config-persistor";
+import { SaleorApiUrl } from "@/modules/saleor/saleor-api-url";
+import { IStripePaymentIntentsApiFactory } from "@/modules/stripe/types";
+
+const saleorApiUrlMock = SaleorApiUrl.create({
+  url: "https://foo.bar.saleor.cloud/graphql/",
+})._unsafeUnwrap();
+const channelIdMock = "test-id";
+const mockAppId = "test-id";
+
+describe("InitializeStripeTransactionUseCase", () => {
+  const testConfig: AppConfigPersistor = {
+    getStripeConfig: vi.fn(),
+    saveStripeConfig: vi.fn(),
+  };
+  const testStripeApiFactory: IStripePaymentIntentsApiFactory = {
+    create: vi.fn(),
+  };
+
+  it("Calls Stripe PaymentIntentsAPI to create payment intent", async () => {
+    vi.mocked(testConfig.getStripeConfig).mockImplementationOnce(async () => mockedStripeConfig);
+    vi.mocked(testStripeApiFactory.create).mockImplementationOnce(
+      () => new MockedStripePaymentIntentsApi(),
+    );
+
+    const spy = vi.spyOn(MockedStripePaymentIntentsApi.prototype, "createPaymentIntent");
```

Comment:
Is accessing prototype save?

I think it's cleaner to 

```
const mockApi = new MockedStripePaymentIntentsApi()

expect(mockApi)....

```

and mock api class should use vi.fn() from the beginning, no need to re-spy it every time